### PR TITLE
Don't enable HC on minigame worlds

### PR DESCRIPTION
**Pull request by aronwk-aaron**
Fix max's ouchi

### CODE REVIEW

**Review by EmosewaMC — DISMISSED** [review]
Couldn't we just check if cloneid is non zero?  If clone id is zero you should be able to respawn in that world

**Comment by aronwk-aaron** [comment]
> Couldn't we just check if cloneid is non zero?  If clone id is zero you should be able to respawn in that world

Yep, that's a better idea

**Review by EmosewaMC — APPROVED** [review]

File: dGame/EntityManager.cpp [hunk 1]
```diff
@@ -69,6 +69,10 @@ void EntityManager::Initialize() {
 	m_HardcoreUscoreEnemiesMultiplier = hcUscoreMult.empty() ? 2 : std::stoi(hcUscoreMult);
 	auto hcDropInv = Game::config->GetValue("hardcore_dropinventory_on_death");
 	m_HardcoreDropinventoryOnDeath = hcDropInv.empty() ? false : (hcDropInv == "1");
+
+	// If instance is not zero, then hardcore mode is disabled
+	// aka minigames and props
+	if (dZoneManager::Instance()->GetZoneID().GetInstanceID() != 0) m_HardcoreMode = false;
```
Comment (Wincent01) [inline dGame/EntityManager.cpp:75]: Believe instance-ID is different from clone-ID

Comment (EmosewaMC) [reply dGame/EntityManager.cpp:75]: ah yes it is.  Glanced right over that 😅

**Review by EmosewaMC — APPROVED** [review]

**Review by Wincent01 — APPROVED** [review]
HC could still be enabled in instances like FV Dragons and Frakjaw, but this catches most cases where is would be a problem.

**Pull request merged**